srml/staking/src/equalize.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Phragmen post-processing fot staking module.
+
+// TODO #3365: deprecate this file, decouple from staking and move to sr_primitives.
+#![cfg(feature = "equalize")]
+
+use crate::{ExpoMap, Trait, BalanceOf, IndividualExposure};
+
+use sr_primitives::traits::{Zero, Convert, Saturating};
+use sr_primitives::phragmen::{ExtendedBalance};
+
+/// Performs equalize post-processing to the output of the election algorithm
+/// This function mutates the input parameters, most noticeably it updates the exposure of
+/// the elected candidates.
+///
+/// No value is returned from the function and the `expo_map` parameter is updated.
+pub fn equalize<T: Trait + 'static>(
```

Comment:
this should be in sr-prtimitives, too. when i said "phragmen", i meant the whole of the phragmen system, not just he initial phrase.

Reply:
It is a bit complicated to do that now. I am refactoring phragmen in a separate branch and if we don't want to equalize for council yet (there is no reason to need it since we are not using edge/weight data; only the list of winners) I prefer doing it later on after my refactor. There's a TODO for it at the top of the file. 

```
// TODO #3365: deprecate this file, decouple from staking and move to sr_primitives.
```

Reply:
this is actually done with the new rebase.